src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -52,7 +105,10 @@
    (comidi/GET "/environments" request
                (request-handler request))
    (comidi/GET ["/status/" [#".*" :rest]] request
-               (request-handler request))))
+               (request-handler request))
+
+   (comidi/GET ["/environment_classes" [#".*" :rest]] request
+               ((environment-class-handler jruby-service) request))))
```

Comment:
Where the functionality for the "environment_classes" endpoint does use a JRuby, another option for this implementation would have been to somehow route it through the `request-handler-service` as is done for the other endpoints under "/puppet/v3".  This endpoint didn't seem to fit that model well, though, in that there's a lot more Clojure-specific functionality planned for it - e.g., use of an internal Clojure/process-wide cache.  Also, at some point, the environment-classes endpoint may not require the use of a JRuby at all if we get to an implementation where the cache is populated asynchronously from an incoming web service request.  The approach in this PR bypasses the `request-handler-service`, while still using the `jruby-puppet-pooled-service` to interact with a JRuby as needed.